### PR TITLE
Result export and fixed bug on time format on poll submit

### PR DESCRIPTION
**Pull request by ewelle**
In this pull request:
    
- Poll result export: On the result tab of a project administrator, there is a download button, that enables you to download the poll results as a csv file.

- bug fixes on delete project: Here the delete button on a project is always available. This was requested by Geron, because project admins must be able to delete a flashpoll, even when the poll is published, and without needing to unpublish it first.
    
- bug fixes on time format and on export function : Taking to account the new date format.

### CODE REVIEW

**Comment by ewelle** [comment]
@MagdaN I added some comments on the PR description. Look it up.

**Comment by slomo** [comment]
Requires #882 to be merged first and than needs merge of master.

File: euth/flashpoll/static/js/flashpoll.js [hunk 1]
```diff
@@ -819,3 +826,117 @@ window.changeType = function(questionType, key, qorderId) {
...
+    //this will remove the blank-spaces from the title and replace it with an underscore
+    fileName += ReportTitle.replace(/ /g,"_");   
+    
+    //Initialize file format you want csv or xls
+    var uri = 'data:text/csv;charset=utf-8,' + escape(CSV);
+    
+    // Now the little tricky part.
+    // you can use either>> window.open(uri);
+    // but this will not work in some browsers
+    // or you will not get the correct file extension
```
Comment (slomo) [inline euth/flashpoll/static/js/flashpoll.js]: better use `window.open()` here

File: euth/flashpoll/static/js/flashpoll.js [hunk 2]
```diff
@@ -819,3 +826,117 @@ window.changeType = function(questionType, key, qorderId) {
...
+    link.style = "visibility:hidden";
+    link.download = fileName + ".csv";
+    
+    //this part will append the anchor tag and remove it after automatic click
+    document.body.appendChild(link);
+    link.click();
+    document.body.removeChild(link);
+}
+
+JSON.flatten = function(data) {
```
Comment (slomo) [inline euth/flashpoll/static/js/flashpoll.js]: Why not use a normal function here.

Comment (slomo) [inline euth/flashpoll/static/js/flashpoll.js]:
```
var flatten =  function(data)  {
   ... 
}
```

File: euth/flashpoll/static/js/flashpoll.js [hunk 3]
```diff
@@ -819,3 +826,117 @@ window.changeType = function(questionType, key, qorderId) {
...
+      for (var i = 0; i < data.length; i++) {
+          var result = {};
+          recurse(data[i], "");
+          list_result[i] = result;
+      }        
+    }
+    return list_result;
+}
+
+function contains(a, obj) {
```
Comment (slomo) [inline euth/flashpoll/static/js/flashpoll.js]: var contains =  function (a, obj) {
...
}

File: euth/flashpoll/templates/euth_flashpoll/includes/project_form_result_flashpoll_tab.html [hunk 4]
```diff
@@ -19,8 +19,14 @@ <h3 class="form-section-heading">{% trans 'Result '%}</h3>
                 <p class='form-help-text'>{{ form.project.result.help_text }}</p>
                 {% render_field form.project.result class="form-control" %}
                 <span class="help-block">{{ error }}</span>
-            </div>
-			{% if pollresult.noOfReceivedResults != 0 %}
+            </div>           
+            {% if poll %}
+                <div style="width: 39px;">
```
Comment (slomo) [inline euth/flashpoll/templates/euth_flashpoll/includes/project_form_result_flashpoll_tab.html]: Could be moved to the _flashpoll.scss

File: euth/flashpoll/templates/euth_flashpoll/includes/project_form_result_flashpoll_tab.html [hunk 5]
```diff
@@ -33,7 +39,7 @@ <h3 class="form-section-heading">{% trans 'Result '%}</h3>
 					</div>
 					{% endif %}
 				{% endfor %}
-			{% endif %}
+			{% endif %}
```
Comment (slomo) [inline euth/flashpoll/templates/euth_flashpoll/includes/project_form_result_flashpoll_tab.html]: Dead whitespace is so dead ;)

**Comment by slomo** [comment]
Instead of doing this in javascript you could also implement a django view instead here.

File: euth/flashpoll/static/js/flashpoll.js (hunk 1 above)
Comment (ewelle) [inline euth/flashpoll/static/js/flashpoll.js]: Apparently, it is not possible to rename the file usint window.open().
The exported file name is always data.csv.
How can we handle this, I do want the file to  be named after the polI.

File: euth/flashpoll/static/js/flashpoll.js (hunk 3 above)
Comment (ewelle) [inline euth/flashpoll/static/js/flashpoll.js]: Done!

File: euth/flashpoll/static/js/flashpoll.js (hunk 2 above)
Comment (ewelle) [inline euth/flashpoll/static/js/flashpoll.js]: Done!

File: euth/flashpoll/templates/euth_flashpoll/includes/project_form_result_flashpoll_tab.html (hunk 4 above)
Comment (ewelle) [inline euth/flashpoll/templates/euth_flashpoll/includes/project_form_result_flashpoll_tab.html]: Done!

File: euth/flashpoll/templates/euth_flashpoll/includes/project_form_result_flashpoll_tab.html (hunk 5 above)
Comment (ewelle) [inline euth/flashpoll/templates/euth_flashpoll/includes/project_form_result_flashpoll_tab.html]: Done!

**Comment by ewelle** [comment]
All the comments have been reviewed and corrected except the one concerning window.open(). See the comments.   

For me this branch is ready to be merged :)

**Pull request merged**